cofact.c

```diff
@@ -981,6 +988,40 @@ int main (int argc, char **argv) {
             exit (1);
```

Comment:
I believe you are missing a `+ 1` here in the Fermat case, as the plus one in the Fermat numbers format requires an extra byte. In other words, I believe it should be `res_len = (1 << (n_proof - 3)) + 1;`, which is constant with [what Mlucas does](https://github.com/primesearch/Mlucas/blob/4db9fbc42ba00298c123623549bd36dd47573d45/src/Mlucas.c#L5199). (This was the only difference I noticed when comparing your implementation to my GIMPS status script.)

Reply:
Technically you are correct that an extra byte is required, in the case that the top most bit is not zero; however that would be the highly unusual case that the $A$ residue $\equiv -1$ (mod $F$). If $F$ is prime it should be the previous iteration’s Pépin residue that should be $-1$, which when squared results in $A \equiv 1$ (mod $F$). I will go out on a limb and suggest every residue likely to ever be imported will have the top most bit equal to zero, so the extra byte is never in practice required.